src/vm/crossbitness.h

```diff
@@ -0,0 +1,46 @@
+#ifndef CROSSBITNESS_H
+#define CROSSBITNESS_H
+
+struct CrossBitness
```

Comment:
I would rather use explicitly manually defined sizes than trust the C++ compiler that it is going to layout the structures the way we expect.

Building upon approach I took in https://github.com/dotnet/coreclr/pull/17876/files .

Reply:
@jkotas I also though about this alternative. But then I would also need to replace all ObjSizeOf(Object) and ObjSizeOf(ArrayBase) expressions with some constants as well. 

Is any standard name for `sizeof(Object) + sizeof(ObjHeader)`?